main/src/ca/uwaterloo/flix/MutationTester.scala

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright 2015-2016 Magnus Madsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.Ast.Constant
+import ca.uwaterloo.flix.language.ast.SemanticOp.{BoolOp, CharOp, Float32Op, Float64Op, Int16Op, Int32Op, Int64Op, Int8Op, StringOp}
+import ca.uwaterloo.flix.language.ast.{Ast, SemanticOp, Symbol, TypedAst}
+import ca.uwaterloo.flix.language.ast.TypedAst.Expr
+import ca.uwaterloo.flix.tools.Tester
+import ca.uwaterloo.flix.util._
+import org.json4s.reflect.fail
+
+import java.io.File
+import scala.::
+import scala.util.Random
+
+object MutationTester {
+    def run(flix: Flix, tester: String, testee: String): Unit = {
+        val root = flix.check().unsafeGet
+        val start = System.nanoTime()
+        val root1 = mutateRoot(root, testee)
+        val end = System.nanoTime() - start
+        val timeSec = end.toFloat / 1_000_000_000.0
+        println(s"time to generate mutations: $timeSec")
+        runMutations(flix, root, root1)
+        /**
+        val result = root1.map(r => flix.codeGen(r).unsafeGet)
+        val tests = result.map(res => res.getTests)
+        */
+    }
+
+    private def runMutations(flix: Flix, root: TypedAst.Root, mutatedDefs: List[(Symbol.DefnSym, List[TypedAst.Def])]): Unit = {
```

Comment:
SWEA: Clean code: One level of abstraction? Do one thing?

Reply:
We have somewhat clean coded this